code/Path_Distance_TableGenerator.py

```diff
@@ -0,0 +1,311 @@
+
+def query_db_for_edges():
+    nodes_query = f"""SELECT * FROM standard_paths;"""
+    results = global_querier.execute_query(nodes_query)
+    for row in results:
+        fc = row[0]
+        fs = row[1]
+        fcc = row[2]
+        tc = row[3]
+        ts = row[4]
+        tcc = row[5]
+        dist_km = float(row[6])
+        path_wkt = row[7]
+        edge = ((fc, fs, fcc), (tc, ts, tcc))
+        global_edges_dict[edge] = {}
+        global_edges_dict[edge]['DIST_KM'] = dist_km
+        global_edges_dict[edge]['GEOM'] = wkt.loads(path_wkt)
+
+def create_graph():
+    # add nodes
+    for n in global_nodes_dict.keys():
+        x = global_nodes_dict[n]['x']
+        y = global_nodes_dict[n]['y']
+        geom = global_nodes_dict[n]['GEOM']
+        global_graph.add_node(n, x=x, y=y, geom=geom)
+    # add edges
+    for e in global_edges_dict.keys():
+        dist = global_edges_dict[e]['DIST_KM']
+        geom = global_edges_dict[e]['GEOM']
+        global_graph.add_edge(e[0], e[1], dist_km=dist, geom=geom)
+
+def initialize_the_global_graph(db_file):
+    global global_querier
+    global_querier = qdb.queryDatabase(db_file)
+    query_db_for_nodes()
+    query_db_for_edges()
+    create_graph()
+
+def get_shortest_path(src, dst, route_geom, waypoints_geom):
+    fc = src.split(',')[0].strip()
+    fs = src.split(',')[1].strip()
+    fcc = src.split(',')[2].strip()
+    tc = dst.split(',')[0].strip()
+    ts = dst.split(',')[1].strip()
+    tcc = dst.split(',')[2].strip()
+
+    src = (fc, fs, fcc)
+    dst = (tc, ts, tcc)
+
+    try:
+        route = nx.shortest_path(global_graph, src, dst, weight='dist_km')
+        for i in range(len(route)-1):
+            e = (route[i], route[i+1])
+            if not e in global_edges_dict.keys():
+                e = (route[i+1], route[i])
+            geom = global_edges_dict[e]["GEOM"]
+            route_geom.append(geom)
+            if not global_nodes_dict[e[0]]["GEOM"] in waypoints_geom:
+                waypoints_geom.append(global_nodes_dict[e[0]]["GEOM"])
+            if not global_nodes_dict[e[1]]["GEOM"] in waypoints_geom:
+                waypoints_geom.append(global_nodes_dict[e[1]]["GEOM"])
+        return route
+    except:
+        print(f"Could not complete query from '{src}' to '{dst}'.", file=sys.stderr)
+        return None
+
+def calc_dist_along_path(route):
+    dist = 0.0
+    for i in range(len(route)-1):
+        e = (route[i], route[i+1])
+        if not e in global_edges_dict.keys():
+            e = (route[i+1], route[i])
+        dist += global_edges_dict[e]["DIST_KM"]
+    return dist
+
+def transform_location_format(location):
+    return location.replace('/', ',')
+
+def find_shortest_path(db_file, start_city, start_state, start_country, end_city, end_state, end_country):
+    # Connect to the SQLite database
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+
+    # cursor.execute('UPDATE phys_nodes AS pnc \
```

Comment:
fixed, a function called update_database has been added to the main function of this file, it will run automatically for each query. The time consumption is within seconds.